daemon/policy.go

```diff
@@ -178,14 +180,58 @@ type AddOptions struct {
 	Source string
 }
 
+// PolicyAddEvent is a wrapper around the parameters for policyAdd.
+type PolicyAddEvent struct {
+	rules policyAPI.Rules
+	opts  *AddOptions
+	d     *Daemon
+}
+
+// Handle implements pkg/eventqueue/EventHandler interface.
+func (p *PolicyAddEvent) Handle(res chan interface{}) {
+	p.d.policyAdd(p.rules, p.opts, res)
```

Comment:
nit: glancing at this, I thought "hmm, this handle function doesn't do anything with the channel", just because it doesn't follow the similar style as the `Handle()` declarations in the endpoint. Then I investigated where the channel is passed to in `p.d.policyAdd()` and discovered it is indeed populated within this call. I guess this is done because the result is handed back before queueing the regeneration. Seems correct, this implementation just shares the responsibility for the channel across more functions.

Reply:
The selective regeneration is mentioned implicitly here, but I will make it more explicit.

Regarding the commit ordering - it has been quite difficult to keep this PR in a manageable state due to the amount of required changes being greater than initially anticipated. In hindsight, I probably should have split it up into another PR, but that's only now knowing the nature of the work needed after having done it. I feel like I will always leave someone unsatisfied with respect to how commits are performed, because if I did the rule-reaction separately, then someone will ask why I am doing it without a general need for it (the answer would be that it would be addressed in a future commit), while if I couple them together to show that one change is needed by the other, then I will get a comment saying that the changes should be split up. 

I feel like keeping the changes co-located based off of package here makes the most sense in terms of keeping the commits maintainable and not ending up in a mess for rebasing. I've spent good chunks of several days rebasing already to address comments :(